src/components/button/Button.tsx

```diff
@@ -300,84 +298,147 @@ export function Button({
-      as: tag,
       ...baseProps,
-      ...(props as AsDivProps),
+      ...(props as ButtonProps<'div'>),
+      tag,
     };
     switch (intent) {
       case 'primary':
-        return <Primary {...buttonProps}>{renderContent('white')}</Primary>;
+        return (
+          <Primary as="div" {...buttonProps}>
+            {renderContent('white')}
+          </Primary>
+        );
       case 'subtle':
-        return <Subtle {...buttonProps}>{renderContent()}</Subtle>;
+        return (
+          <Subtle as="div" {...buttonProps}>
+            {renderContent()}
+          </Subtle>
+        );
       case 'outline':
-        return <Outline {...buttonProps}>{renderContent()}</Outline>;
+        return (
+          <Outline as="div" {...buttonProps}>
+            {renderContent()}
+          </Outline>
+        );
       case 'warning':
-        return <Warning {...buttonProps}>{renderContent()}</Warning>;
+        return (
+          <Warning as="div" {...buttonProps}>
+            {renderContent()}
+          </Warning>
+        );
       case 'danger':
-        return <Danger {...buttonProps}>{renderContent()}</Danger>;
+        return (
+          <Danger as="div" {...buttonProps}>
+            {renderContent()}
+          </Danger>
+        );
       case 'text':
-        return <TextButton {...buttonProps}>{renderContent()}</TextButton>;
+        return (
+          <TextButton as="div" {...buttonProps}>
+            {renderContent()}
+          </TextButton>
+        );
       case 'link':
-        return <LinkButton {...buttonProps}>{renderContent()}</LinkButton>;
+        return (
+          <LinkButton as="div" {...buttonProps}>
+            {renderContent()}
+          </LinkButton>
+        );
       case 'secondary':
       default:
-        return <Secondary {...buttonProps}>{renderContent()}</Secondary>;
+        return (
+          <Secondary as="div" {...buttonProps}>
+            {renderContent()}
+          </Secondary>
+        );
     }
   }
-  if (tag === 'a') {
-    const buttonProps = {
-      as: tag,
-      ...baseProps,
-      ...(props as AsAnchorProps),
-    };
-    switch (intent) {
-      case 'primary':
-        return <Primary {...buttonProps}>{renderContent('white')}</Primary>;
-      case 'subtle':
-        return <Subtle {...buttonProps}>{renderContent()}</Subtle>;
-      case 'outline':
-        return <Outline {...buttonProps}>{renderContent()}</Outline>;
-      case 'warning':
-        return <Warning {...buttonProps}>{renderContent()}</Warning>;
-      case 'danger':
-        return <Danger {...buttonProps}>{renderContent()}</Danger>;
-      case 'text':
-        return <TextButton {...buttonProps}>{renderContent()}</TextButton>;
-      case 'link':
-        return <LinkButton {...buttonProps}>{renderContent()}</LinkButton>;
-      case 'secondary':
-      default:
-        return <Secondary {...buttonProps}>{renderContent()}</Secondary>;
-    }
+
+  const buttonProps = {
+    type,
```

Comment:
```suggestion
```

Reply:
I think the reason for this is to overwrite the default type 'submit' to 'button' if type is not passed in

Reply:
My bad. Nice catch 😅